controllers/Backend.php

```diff
@@ -1,104 +1,344 @@
+            if(empty($errors)){
+                Usermanager::update_password($password);
+                header("Location:/modification-du-mot-de-passe");
+            }
+        }else
+            $errors =[];
         
-        $page="backend/password";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        if($manager_user->hasnt_password() == 0){
+        header("Location:/dashboard");
+    }
+        $myView = new View('password', 'backend');
+        $myView->render(array('errors' => $errors));
     }
 
     public function postback()
     {
-        $model_article = new Article();
-        $model_user = new User();
+        if(isset($_GET['id']))
+        {
+        $id = $_GET['id'];
+        $manager_article = new ArticleManager();
+        $posts = $manager_article->get_post();
+        }
+
+          if(filter_has_var(INPUT_POST, 'delete')){
+            $manager_article->delete_article_comments();
+            $manager_article->delete_article();
+        }
+
+        if(filter_has_var(INPUT_POST, 'submit')){
+            if(filter_has_var(INPUT_POST, 'title')){
+                $title = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'title')));
+            }
+            if(filter_has_var(INPUT_POST, 'chapo')){
+                $chapo = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'chapo')));
+            }
+            if(filter_has_var(INPUT_POST, 'content')){
+                $content = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'content')));
+            }
+            $posted = filter_input(INPUT_POST, 'public') ? "1" : "0";
+            
+            $errors = [];
+
+            if(empty($title) || empty($content) || empty($chapo)){
+                $errors['empty'] = "Veuillez remplir tous les champs svp";
+            }
+            if(strlen($title) < 5){
+                $errors['title'] = "Votre message doit contenir au moins 5 caractères.";
+            }
+            if(strlen($chapo) < 5){
+                $errors['chapo'] = "Votre chapô doit contenir au moins 5 caractères.";
+            }
+            if(strlen($content) < 5){
+                $errors['content'] = "Votre article doit contenir au moins 5 caractères.";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+  
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable.";
+                }
+            }
+            
+            if(empty($errors)){
+                $manager_article->edit($title,$chapo,$content,$posted, filter_input(INPUT_GET, 'id'));
+             
+                if(!empty($_FILES['image']['name']))
+                {
+                    $manager_article->update_img($_FILES['image']['tmp_name'], $extension);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Direct use of $_FILES Superglobal detected.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)